lib/dma.h

```diff
@@ -82,6 +82,12 @@
 
 #define iov_end(iov) ((iov)->iov_base + (iov)->iov_len)
 
+/* Based on QEMU's atomic primitives. */
```

Comment:
move to lib/private.h with suitable prefix?

Also, I think at least having the comment as to why we need dma_barrier() is worth having too

Reply:
ACK

Reply:
Done